### ISSUE TITLE
Sample Tensor Operations

### ISSUE THREAD

**Issue by hmustafajr** [hmustafajr/PyTorch-Fundamentals#2]
numpy-esque Indexing and Slicing

https://github.com/hmustafajr/Python-Toolbox/blob/main/numpy-esque%20Indexing%20and%20Slicing.py

Joining tensors
https://github.com/hmustafajr/Python-Toolbox/blob/main/Joining%20Tensors.py

torh.cat - can be used to concatenate a sequence of tensors along a given dimension.

torch.stack - is a related tensor joining option that concatenates a sequence of tensors along a new dimension.

Ex:
t1 = torch.cat([tensor, tensor, tensor], dim=1)
print(t1)

Arithmetic Operations
https://github.com/hmustafajr/Python-Toolbox/blob/main/Arithmetic%20Operations.py

Single Element Tensors
https://github.com/hmustafajr/Python-Toolbox/blob/main/Single%20Element%20Tensors.py

If you have a one element tensor, say by aggregating all values of a tensor into one value, you can convert it to a Python numerical value using item()

In-place Operations
https://github.com/hmustafajr/Python-Toolbox/blob/main/In-place%20Operations.py

Operations that store the result into the operand are called in-place. They are denoted by an _ suffix. For example: x.copy_(y), x.t_(), will change x.

Note: In-place operations save some memory, but can be really problematic when computing derivatives because of their immediate loss of history.

Bridge w/ NumPy
CPU tensors and NumPy arrays can share their underlying memory locations, and changing one will change the other.

- Tensor to NumPy array
https://github.com/hmustafajr/Python-Toolbox/blob/main/Tensor%20to%20NumPy%20array.py

- NumPy array to tensor
https://github.com/hmustafajr/Python-Toolbox/blob/main/NumPy%20array%20to%20tensor.py

What is true about Tensors?
Tensors are specialized data structures that are similar to arrays and matrices.
